core/src/main/scala/flatgraph/storage/Deserialization.scala

```diff
@@ -185,21 +186,25 @@ object Deserialization {
 
     val headerBytes = new Array[Byte](Keys.Header.length)
     header.get(headerBytes)
-    if (!Arrays.equals(headerBytes, Keys.Header))
+    if (!Arrays.equals(headerBytes, Keys.Header)) {
       throw new DeserializationException(
         s"expected header '$MagicBytesString' (`${Keys.Header.mkString("")}`), but found '${headerBytes.mkString("")}'"
       )
+    }
 
     val manifestOffset = header.getLong()
     val manifestSize   = channel.size() - manifestOffset
-    val manifestBytes  = ByteBuffer.allocate(manifestSize.toInt)
+    if (manifestSize > fileSize) {
+      throw new DeserializationException(s"corrupt file: manifest size ($manifestSize) cannot be larger than the file's size ($fileSize)")
+    }
+
+    val manifestBytes = ByteBuffer.allocate(manifestSize.toInt)
```

Comment:
If you want to do this check at all, then please also handle the case where manifestSize overflows the 32 bit integer.

Reply:
👍🏻 doen